puppetboard/docker_settings.py

```diff
@@ -1,4 +1,34 @@
 import os
+import tempfile
+import base64
+import binascii
+
+
+def cert_to_file(cert_file_or_string):
+    """
+    cert_to_file takes in a string, if it looks like a certificate, save the contents to a temporary
+    file and return the path to that temporary file
+
+    cert_to_file also supports base64 encoded certificates, in cases where newlines cause problems
+
+    Note: NamedTemporaryFile does not work great with Windows, but since this is for Docker
+    hopefully we'll be fine.
+    """
+    if isinstance(cert_file_or_string, str):
+        try:
+            cert_str = base64.b64decode(cert_file_or_string).decode()
+        except (UnicodeDecodeError, binascii.Error):
+            cert_str = cert_file_or_string
+
+        if '-----BEGIN' in cert_str:
+            with tempfile.NamedTemporaryFile(delete=False, suffix='puppetboard_cert') as tmpfile:
+                tmpfile.write(cert_str.encode())
```

Comment:
do you want to be removing the `-----BEGIN CERTIFICATE-----` from the passed `cert_str` before you encode it? 
from the README, i kinda thought that the BEGIN and END would just be identifiers, but dropped by the time you used the actual cert.

Reply:
We need to keep the BEGIN and END as they are part of the [PEM format](https://knowledge.digicert.com/quovadis/ssl-certificates/ssl-general-topics/what-is-pem-format.html) for cert/chain/key that the Requests library expects.